x-pack/plugins/observability_ai_assistant/public/components/chat/chat_body.tsx

```diff
@@ -303,6 +307,48 @@ export function ChatBody({
                           })
                         );
                         break;
+                      case ChatActionClickType.updateVisualization:
+                        const visualizeQueryMessagesIndex = messages.findIndex(
+                          ({ message }) => message.name === 'visualize_query'
+                        );
+                        next(
+                          messages.slice(0, visualizeQueryMessagesIndex).concat({
+                            '@timestamp': new Date().toISOString(),
+                            message: {
+                              role: MessageRole.Assistant,
+                              content: '',
+                              function_call: {
+                                name: 'visualize_query',
+                                arguments: JSON.stringify({
+                                  query: payload.query,
+                                  newInput: payload.newInput,
```

Comment:
this is about user overrides right? can we name it as such?

Reply:
Done https://github.com/elastic/kibana/pull/174677/commits/a72b140f7a48d8dfa89fa96ca5414ec3d34ea226